### PR TITLE
#165413121 user can bookmark article to read later

### PR DESCRIPTION
**Pull request by Kabalisa**
#### Description
users who read articles on authors haven may appreciate the articles very much that they would want to read them later. this feature will help them bookmark them to make that easy.
#### Type of change
N/A
#### How Has This Been Tested?
this has been tested by sending request specified in the readme to these endpoints
*  create a bookmark on an article ```POST  /api/v1/articles/{slug}/bookmark```
*  get bookmarks on an article ```GET  /api/v1/articles/user/bookmarks```
#### Checklist:
N/A
#### Pivotal Tracker ID
[#165413121](https://www.pivotaltracker.com/story/show/165413121)

### CODE REVIEW

File: tests/bookmark.test.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+//         res.body.should.be.a('object');
+//         res.body.should.have.property('Bookmarks');
+//         res.body.Bookmarks[0].should.have.property('id');
+//         res.body.Bookmarks[0].should.have.property('userId');
+//         res.body.Bookmarks[0].should.have.property('createdAt');
+//         res.body.Bookmarks[0].should.have.property('article');
+//         done();
+//       });
+//   });
+// });
```
Comment (ericnyirimana) [inline tests/bookmark.test.js]: Remove these comments

**Review by jabichris — COMMENTED** [review]
I tested it locally and everything is working well. But I have an issue with using `user` in `/api/v1/articles/user/bookmarks` . aren't there any other way of getting bookmarks?

**Comment by Kabalisa** [comment]
> I tested it locally and everything is working well. But I have an issue with using `user` in `/api/v1/articles/user/bookmarks` . aren't there any other way of getting bookmarks?

user is there for readability of the endpoint. it means the bookmarks fetched are for a single user and that user is the logged in user. the bookmarks are fetched using the user id from the token.

File: tests/bookmark.test.js (hunk 1 above)
Comment (Kabalisa) [reply tests/bookmark.test.js]: done

**Comment by istafrica-dennw** [comment]
you can remove the /user in bookmarks url

**Pull request merged**